relay-server/src/actors/spooler/mod.rs

```diff
@@ -345,7 +349,19 @@ impl OnDisk {
             // to pick up.
             //
             // Right now we use 100 for batch size.
-            let mut envelopes = sql::delete_and_fetch(key, 100).fetch(&self.db).peekable();
+            let batch_size = 100;
```

Comment:
nit: We might want to adjust the batch size (currently we hardcode `100`) in case if the configured buffer guards number is actually smaller than the batch.